pulsar-client/src/main/java/org/apache/pulsar/client/impl/TypedMessageBuilderImpl.java

```diff
@@ -130,6 +131,32 @@ public MessageId send() throws PulsarClientException {
         return this;
     }
 
+    @SuppressWarnings("unchecked")
+    @Override
+    public TypedMessageBuilder<T> loadConf(Map<String, Object> config) {
+        config.forEach((key, value) -> {
+            if (key.equals("key")) {
+                this.key(checkType(value, String.class));
+            } else if (key.equals("properties")) {
+                this.properties(checkType(value, Map.class));
+            } else if (key.equals("eventTime")) {
+                this.eventTime(checkType(value, Number.class).longValue());
```

Comment:
why it is `Number.class` not `Long.class`?

Reply:
If it's `Long`, then you'd have to write: 

```java
conf.put("eventTime", 12346L);
```

With number it will accept also an `int`, in the same way you can pass `int` to a function that takes a `long` parameter

Reply:
but a float or double is also number, right?

Reply:
Yes, it will get truncated

Reply:
@sijie  anyway, changed it to Long